### DISCUSSION TITLE
new notes: slider mid-notes and mini-spinners

### DISCUSSION THREAD

**Discussion by ghost** [ppy/osu#18867] [Ideas] [answered]
I know my idea is going to get rejected as it's throwing a tad bit of complication onto an extremely simple game and im asking for new notes that provide exceptions to General Rule No.1 of Ranking Criteria that states no two objects can be on any 1 tick, but I can't help but not share my unwanted thoughts.

Slider mid-notes
---
Notes that occur during a slider, they are locked to the curve of the slider and cannot be placed outside of it

Purpose:
- To better represent legato and slur notes, the standard mapping method used today is to end one a 1/4 beat early with no hit sound and start a new slider OR to create harsh movement within the sliderbody. the former is just an inferior version of the idea I propose here and the latter could be used to indicate softer changes in the notes whereas my idea would be used for the more prominent ones.
- AKA: better/more options for mappers

Player:
- Player must swap the key they are using to hold the slider in order to hit a mid-note. Should they release one key and then press the other then a slider break would ensue, they must press button 2 while still holding down button 1, then release button 1. The game will make an exception for this 1 frame sliderbreak.

Mapping:
- Create a circle while you have a slider selected, it will automatically place itself where it belongs at the mid-point between the two furthest parts (editor calculates the distances between each slider-head, mid-note, and slider-end and chooses the gap with the most space to place the cirlce, in event of tie; utilize the first gap)
- the mapper can then click and drag on said circle to move it along the slider (it'll snap according to the timing)
- Should the mapper desire, they can set a red anchor on the circle itself by ctrl clicking the circle, now the mapper may move the circle outside the domain of the already set slider.
- placing a circle on a reverse slider will result in a one time press on the first traversal along the slider-body, the mapper may click and drag the circle in the timeline to make it jump to a different section of the reverse slider
- placing a circle with a red anchor on a reverse slider applies a permanent circle to each traversal of the sliderbody

Side notes:
- Placing a mid-note on a slider should reset the slider ticks

Mini-spinners
---
A spinner that revolves around a specific note, playing immediately after it's been pressed.

Purpose:
- Variety of options for the mapper to utilize

- For powerful non-staccato notes, the answer is almost always to use a fast slider or a spinner to mimic the intensity, now they could use a mini spinner to indicate a powerful note. Since the spin is focused on the note itself, it'll (probably) for more connected to the line in the song, as opposed to going back to center for a spinner

- Standard spinners will still be used more for a winding up portion of the song, while these mini-spinners will take over some of the powerful notes that used to be taken by spinners and fast sliders

- increase the skill cap of a song's max score (spinners are usually the last thing to optimize in getting a top score), allowing for scores to take even longer to cap out

Player:
- Press a note that has a spinner icon around it and then spins around the note they pressed (not the center like a standard spinner)
- This spinner should take much less time to succeed and give less bonus points (50%?) (these changes are because: A) its most commonly going to be used for shorter held notes, and B) since a mini-spinner will be easier to complete, it shouldn't reward as much as a standard spinner, (they are MINI-spinners after all))

Mapper:
- Have a circle overlap the first beat of a spinner in the timeline, the spinner should automatically center itself on the note.

Side-notes:
- A mapper should be able to put a mini-spinner as a mid-note, assuming they place the mid-note ON the slider-end. Shouldn't really need to program a way to stop it as General Rule No.1 of Ranking Criteria is to not place anything on the same tick

**Comment by peppy** [comment, marked as answer]
Both of these are already planned, likely in a slightly different way to how you imagine. I'm not intending to add them until everything else is in a good state, but you can look forward to them for now!

**Reply by ghost** [reply]
awesome, thank you for letting me know! I look forward to the day.
